GC overhead limit exceed Exception while running the app

> run
[info] Processing resources
[info] QueuedCruncher is using C:\Users\DELL\AppData\Local\Android\Sdk\build-tools\27.0.0\aapt.exe
ProGuard, version 5.0
ProGuard is released under the GNU General Public License. You therefore
must ensure that programs that link to it (android, ...)
carry the GNU General Public License as well. Alternatively, you can
apply for an exception with the author of ProGuard.
Reading input...
Reading program jar [E:\project\hello-scala-master\src\main\libs\android-support-v4.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-appcompat-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-v4-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-media-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-utils-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-ui-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-fragment-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-animated-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-cardview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-design-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-recyclerview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-transition-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-gridlayout-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-library\jars\scala-library-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-builder_2.11\jars\communicator-builder_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-request_2.11\jars\communicator-request_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okhttp3\okhttp\jars\okhttp-3.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okio\okio\jars\okio-1.13.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-eval_2.11\jars\monix-eval_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-types_2.11\jars\monix-types_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-execution_2.11\jars\monix-execution_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.reactivestreams\reactive-streams\jars\reactive-streams-1.0.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-core_2.11\jars\cats-core_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-macros_2.11\jars\cats-macros_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.github.mpilquist\simulacrum_2.11\jars\simulacrum_2.11-0.10.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\macro-compat_2.11\jars\macro-compat_2.11-1.1.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\machinist_2.11\jars\machinist_2.11-0.6.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-reflect\jars\scala-reflect-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-kernel_2.11\jars\cats-kernel_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-websocket_2.11\jars\communicator-websocket_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-reactive_2.11\jars\monix-reactive_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.jctools\jctools-core\bundles\jctools-core-2.0.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.slf4j\slf4j-api\jars\slf4j-api-1.7.25.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-phoenix_2.11\jars\communicator-phoenix_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-parser_2.11\jars\circe-parser_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-jawn_2.11\jars\circe-jawn_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-core_2.11\jars\circe-core_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-numbers_2.11\jars\circe-numbers_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.spire-math\jawn-parser_2.11\jars\jawn-parser_2.11-0.10.4.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\phoenix-models_2.11\jars\phoenix-models_2.11-1.0.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-generic_2.11\jars\circe-generic_2.11-0.7.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.chuusai\shapeless_2.11\bundles\shapeless_2.11-2.3.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\sdk\extras\android\m2repository\com\android\support\support-annotations\25.3.1\support-annotations-25.3.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.google.code.findbugs\jsr305\jars\jsr305-3.0.2.jar] (filtered)
Reading program jar [E:\project\hello-scala-master\target\android\intermediates\classes.jar] (filtered)
Reading library jar [C:\Users\DELL\AppData\Local\Android\Sdk\platforms\android-26\android.jar]
Initializing...
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at sbt.ConcurrentRestrictions$$anon$4.take(ConcurrentRestrictions.scala:188)
        at sbt.Execute.next$1(Execute.scala:85)
        at sbt.Execute.processAll(Execute.scala:88)
        at sbt.Execute.runKeep(Execute.scala:68)
        at sbt.EvaluateTask$.liftedTree1$1(EvaluateTask.scala:359)
        at sbt.EvaluateTask$.run$1(EvaluateTask.scala:358)
        at sbt.EvaluateTask$.runTask(EvaluateTask.scala:378)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:69)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:67)
        at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:314)
        at sbt.Aggregation$.timedRun(Aggregation.scala:67)
        at sbt.Aggregation$.runTasks(Aggregation.scala:76)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:117)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:115)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:253)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:250)
        at sbt.Command$.process(Command.scala:93)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.State$$anon$1.doX$1(State.scala:183)
        at sbt.State$$anon$1.process(State.scala:190)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.MainLoop$.next(MainLoop.scala:96)
        at sbt.MainLoop$.run(MainLoop.scala:89)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
        at sbt.Using.apply(Using.scala:24)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
        at sbt.MainLoop$.runLogged(MainLoop.scala:22)
        at sbt.StandardMain$.runManaged(Main.scala:109)
        at sbt.xMain.run(Main.scala:38)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:41)
        at xsbt.boot.Boot$.main(Boot.scala:17)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.lang.String.substring(Unknown Source)
        at proguard.classfile.util.DescriptorClassEnumeration.nextClassName(DescriptorClassEnumeration.java:204)
        at proguard.classfile.util.ClassReferenceInitializer.findReferencedClasses(ClassReferenceInitializer.java:510)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramMethod(ClassReferenceInitializer.java:134)
        at proguard.classfile.ProgramMethod.accept(ProgramMethod.java:71)
        at proguard.classfile.ProgramClass.methodsAccept(ProgramClass.java:504)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramClass(ClassReferenceInitializer.java:104)
        at proguard.classfile.ProgramClass.accept(ProgramClass.java:346)
        at proguard.classfile.ClassPool.classesAccept(ClassPool.java:124)
        at proguard.Initializer.execute(Initializer.java:146)
        at proguard.ProGuard.initialize(ProGuard.java:222)
        at proguard.ProGuard.execute(ProGuard.java:97)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at android.Proguard$.runProguard(proguard.scala:210)
        at android.Proguard$.proguard(proguard.scala:164)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1062)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1060)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:35)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[error] java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
[error] Use 'last' for the full log.
> run
[info] Processing resources
[info] QueuedCruncher is using C:\Users\DELL\AppData\Local\Android\Sdk\build-tools\27.0.0\aapt.exe
ProGuard, version 5.0
ProGuard is released under the GNU General Public License. You therefore
must ensure that programs that link to it (android, ...)
carry the GNU General Public License as well. Alternatively, you can
apply for an exception with the author of ProGuard.
Reading input...
Reading program jar [E:\project\hello-scala-master\src\main\libs\android-support-v4.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-appcompat-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-v4-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-media-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-utils-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-ui-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-fragment-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-animated-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-cardview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-design-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-recyclerview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-transition-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-gridlayout-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-library\jars\scala-library-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-builder_2.11\jars\communicator-builder_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-request_2.11\jars\communicator-request_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okhttp3\okhttp\jars\okhttp-3.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okio\okio\jars\okio-1.13.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-eval_2.11\jars\monix-eval_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-types_2.11\jars\monix-types_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-execution_2.11\jars\monix-execution_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.reactivestreams\reactive-streams\jars\reactive-streams-1.0.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-core_2.11\jars\cats-core_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-macros_2.11\jars\cats-macros_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.github.mpilquist\simulacrum_2.11\jars\simulacrum_2.11-0.10.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\macro-compat_2.11\jars\macro-compat_2.11-1.1.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\machinist_2.11\jars\machinist_2.11-0.6.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-reflect\jars\scala-reflect-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-kernel_2.11\jars\cats-kernel_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-websocket_2.11\jars\communicator-websocket_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-reactive_2.11\jars\monix-reactive_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.jctools\jctools-core\bundles\jctools-core-2.0.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.slf4j\slf4j-api\jars\slf4j-api-1.7.25.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-phoenix_2.11\jars\communicator-phoenix_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-parser_2.11\jars\circe-parser_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-jawn_2.11\jars\circe-jawn_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-core_2.11\jars\circe-core_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-numbers_2.11\jars\circe-numbers_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.spire-math\jawn-parser_2.11\jars\jawn-parser_2.11-0.10.4.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\phoenix-models_2.11\jars\phoenix-models_2.11-1.0.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-generic_2.11\jars\circe-generic_2.11-0.7.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.chuusai\shapeless_2.11\bundles\shapeless_2.11-2.3.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\sdk\extras\android\m2repository\com\android\support\support-annotations\25.3.1\support-annotations-25.3.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.google.code.findbugs\jsr305\jars\jsr305-3.0.2.jar] (filtered)
Reading program jar [E:\project\hello-scala-master\target\android\intermediates\classes.jar] (filtered)
Reading library jar [C:\Users\DELL\AppData\Local\Android\Sdk\platforms\android-26\android.jar]
Initializing...
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at sbt.ConcurrentRestrictions$$anon$4.take(ConcurrentRestrictions.scala:188)
        at sbt.Execute.next$1(Execute.scala:85)
        at sbt.Execute.processAll(Execute.scala:88)
        at sbt.Execute.runKeep(Execute.scala:68)
        at sbt.EvaluateTask$.liftedTree1$1(EvaluateTask.scala:359)
        at sbt.EvaluateTask$.run$1(EvaluateTask.scala:358)
        at sbt.EvaluateTask$.runTask(EvaluateTask.scala:378)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:69)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:67)
        at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:314)
        at sbt.Aggregation$.timedRun(Aggregation.scala:67)
        at sbt.Aggregation$.runTasks(Aggregation.scala:76)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:117)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:115)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:253)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:250)
        at sbt.Command$.process(Command.scala:93)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.State$$anon$1.doX$1(State.scala:183)
        at sbt.State$$anon$1.process(State.scala:190)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.MainLoop$.next(MainLoop.scala:96)
        at sbt.MainLoop$.run(MainLoop.scala:89)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
        at sbt.Using.apply(Using.scala:24)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
        at sbt.MainLoop$.runLogged(MainLoop.scala:22)
        at sbt.StandardMain$.runManaged(Main.scala:109)
        at sbt.xMain.run(Main.scala:38)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:41)
        at xsbt.boot.Boot$.main(Boot.scala:17)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.lang.String.substring(Unknown Source)
        at proguard.classfile.util.DescriptorClassEnumeration.nextFluff(DescriptorClassEnumeration.java:177)
        at proguard.classfile.util.ClassReferenceInitializer.findReferencedClasses(ClassReferenceInitializer.java:509)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramMethod(ClassReferenceInitializer.java:134)
        at proguard.classfile.ProgramMethod.accept(ProgramMethod.java:71)
        at proguard.classfile.ProgramClass.methodsAccept(ProgramClass.java:504)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramClass(ClassReferenceInitializer.java:104)
        at proguard.classfile.ProgramClass.accept(ProgramClass.java:346)
        at proguard.classfile.ClassPool.classesAccept(ClassPool.java:124)
        at proguard.Initializer.execute(Initializer.java:146)
        at proguard.ProGuard.initialize(ProGuard.java:222)
        at proguard.ProGuard.execute(ProGuard.java:97)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at android.Proguard$.runProguard(proguard.scala:210)
        at android.Proguard$.proguard(proguard.scala:164)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1062)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1060)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:35)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
[error] java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
[error] Use 'last' for the full log.
> run
[info] Processing resources
[info] QueuedCruncher is using C:\Users\DELL\AppData\Local\Android\Sdk\build-tools\27.0.0\aapt.exe
ProGuard, version 5.0
ProGuard is released under the GNU General Public License. You therefore
must ensure that programs that link to it (android, ...)
carry the GNU General Public License as well. Alternatively, you can
apply for an exception with the author of ProGuard.
Reading input...
Reading program jar [E:\project\hello-scala-master\src\main\libs\android-support-v4.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-appcompat-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-v4-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-media-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-utils-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-ui-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-fragment-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-animated-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-cardview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-design-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-recyclerview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-transition-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-gridlayout-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-library\jars\scala-library-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-builder_2.11\jars\communicator-builder_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-request_2.11\jars\communicator-request_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okhttp3\okhttp\jars\okhttp-3.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okio\okio\jars\okio-1.13.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-eval_2.11\jars\monix-eval_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-types_2.11\jars\monix-types_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-execution_2.11\jars\monix-execution_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.reactivestreams\reactive-streams\jars\reactive-streams-1.0.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-core_2.11\jars\cats-core_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-macros_2.11\jars\cats-macros_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.github.mpilquist\simulacrum_2.11\jars\simulacrum_2.11-0.10.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\macro-compat_2.11\jars\macro-compat_2.11-1.1.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\machinist_2.11\jars\machinist_2.11-0.6.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-reflect\jars\scala-reflect-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-kernel_2.11\jars\cats-kernel_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-websocket_2.11\jars\communicator-websocket_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-reactive_2.11\jars\monix-reactive_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.jctools\jctools-core\bundles\jctools-core-2.0.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.slf4j\slf4j-api\jars\slf4j-api-1.7.25.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-phoenix_2.11\jars\communicator-phoenix_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-parser_2.11\jars\circe-parser_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-jawn_2.11\jars\circe-jawn_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-core_2.11\jars\circe-core_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-numbers_2.11\jars\circe-numbers_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.spire-math\jawn-parser_2.11\jars\jawn-parser_2.11-0.10.4.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\phoenix-models_2.11\jars\phoenix-models_2.11-1.0.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-generic_2.11\jars\circe-generic_2.11-0.7.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.chuusai\shapeless_2.11\bundles\shapeless_2.11-2.3.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\sdk\extras\android\m2repository\com\android\support\support-annotations\25.3.1\support-annotations-25.3.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.google.code.findbugs\jsr305\jars\jsr305-3.0.2.jar] (filtered)
Reading program jar [E:\project\hello-scala-master\target\android\intermediates\classes.jar] (filtered)
Reading library jar [C:\Users\DELL\AppData\Local\Android\Sdk\platforms\android-26\android.jar]
Initializing...
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at sbt.ConcurrentRestrictions$$anon$4.take(ConcurrentRestrictions.scala:188)
        at sbt.Execute.next$1(Execute.scala:85)
        at sbt.Execute.processAll(Execute.scala:88)
        at sbt.Execute.runKeep(Execute.scala:68)
        at sbt.EvaluateTask$.liftedTree1$1(EvaluateTask.scala:359)
        at sbt.EvaluateTask$.run$1(EvaluateTask.scala:358)
        at sbt.EvaluateTask$.runTask(EvaluateTask.scala:378)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:69)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:67)
        at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:314)
        at sbt.Aggregation$.timedRun(Aggregation.scala:67)
        at sbt.Aggregation$.runTasks(Aggregation.scala:76)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:117)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:115)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:253)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:250)
        at sbt.Command$.process(Command.scala:93)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.State$$anon$1.doX$1(State.scala:183)
        at sbt.State$$anon$1.process(State.scala:190)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.MainLoop$.next(MainLoop.scala:96)
        at sbt.MainLoop$.run(MainLoop.scala:89)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
        at sbt.Using.apply(Using.scala:24)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
        at sbt.MainLoop$.runLogged(MainLoop.scala:22)
        at sbt.StandardMain$.runManaged(Main.scala:109)
        at sbt.xMain.run(Main.scala:38)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:41)
        at xsbt.boot.Boot$.main(Boot.scala:17)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.lang.String.substring(Unknown Source)
        at proguard.classfile.util.DescriptorClassEnumeration.nextFluff(DescriptorClassEnumeration.java:177)
        at proguard.classfile.util.DescriptorClassEnumeration.classCount(DescriptorClassEnumeration.java:69)
        at proguard.classfile.util.ClassReferenceInitializer.findReferencedClasses(ClassReferenceInitializer.java:500)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramMethod(ClassReferenceInitializer.java:134)
        at proguard.classfile.ProgramMethod.accept(ProgramMethod.java:71)
        at proguard.classfile.ProgramClass.methodsAccept(ProgramClass.java:504)
        at proguard.classfile.util.ClassReferenceInitializer.visitProgramClass(ClassReferenceInitializer.java:104)
        at proguard.classfile.ProgramClass.accept(ProgramClass.java:346)
        at proguard.classfile.ClassPool.classesAccept(ClassPool.java:124)
        at proguard.Initializer.execute(Initializer.java:146)
        at proguard.ProGuard.initialize(ProGuard.java:222)
        at proguard.ProGuard.execute(ProGuard.java:97)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
        at java.lang.reflect.Method.invoke(Unknown Source)
        at android.Proguard$.runProguard(proguard.scala:210)
        at android.Proguard$.proguard(proguard.scala:164)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1062)
        at android.Tasks$$anonfun$86.apply(tasks.scala:1060)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:35)
        at scala.Function9$$anonfun$tupled$1.apply(Function9.scala:34)
        at scala.Function1$$anonfun$compose$1.apply(Function1.scala:47)
        at sbt.$tilde$greater$$anonfun$$u2219$1.apply(TypeFunctions.scala:40)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
[error] java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
[error] Use 'last' for the full log.
> run
[info] Processing resources
[info] QueuedCruncher is using C:\Users\DELL\AppData\Local\Android\Sdk\build-tools\27.0.0\aapt.exe
ProGuard, version 5.0
ProGuard is released under the GNU General Public License. You therefore
must ensure that programs that link to it (android, ...)
carry the GNU General Public License as well. Alternatively, you can
apply for an exception with the author of ProGuard.
Reading input...
Reading program jar [E:\project\hello-scala-master\src\main\libs\android-support-v4.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-appcompat-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-v4-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-media-compat-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-utils-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-core-ui-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-fragment-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-support-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-animated-vector-drawable-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-cardview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-design-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-recyclerview-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-transition-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\exploded-aars\com.android.support-gridlayout-v7-25.3.1\classes.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-library\jars\scala-library-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-builder_2.11\jars\communicator-builder_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-request_2.11\jars\communicator-request_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okhttp3\okhttp\jars\okhttp-3.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.squareup.okio\okio\jars\okio-1.13.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-eval_2.11\jars\monix-eval_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-types_2.11\jars\monix-types_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-execution_2.11\jars\monix-execution_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.reactivestreams\reactive-streams\jars\reactive-streams-1.0.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-core_2.11\jars\cats-core_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-macros_2.11\jars\cats-macros_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.github.mpilquist\simulacrum_2.11\jars\simulacrum_2.11-0.10.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\macro-compat_2.11\jars\macro-compat_2.11-1.1.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\machinist_2.11\jars\machinist_2.11-0.6.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.scala-lang\scala-reflect\jars\scala-reflect-2.11.11.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.typelevel\cats-kernel_2.11\jars\cats-kernel_2.11-0.9.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-websocket_2.11\jars\communicator-websocket_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.monix\monix-reactive_2.11\jars\monix-reactive_2.11-2.3.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.jctools\jctools-core\bundles\jctools-core-2.0.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.slf4j\slf4j-api\jars\slf4j-api-1.7.25.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\communicator-phoenix_2.11\jars\communicator-phoenix_2.11-3.3.0-RC1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-parser_2.11\jars\circe-parser_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-jawn_2.11\jars\circe-jawn_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-core_2.11\jars\circe-core_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-numbers_2.11\jars\circe-numbers_2.11-0.8.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\org.spire-math\jawn-parser_2.11\jars\jawn-parser_2.11-0.10.4.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.taig\phoenix-models_2.11\jars\phoenix-models_2.11-1.0.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\io.circe\circe-generic_2.11\jars\circe-generic_2.11-0.7.0.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.chuusai\shapeless_2.11\bundles\shapeless_2.11-2.3.2.jar] (filtered)
Reading program jar [C:\Users\DELL\.android\sbt\sdk\extras\android\m2repository\com\android\support\support-annotations\25.3.1\support-annotations-25.3.1.jar] (filtered)
Reading program jar [C:\Users\DELL\.ivy2\cache\com.google.code.findbugs\jsr305\jars\jsr305-3.0.2.jar] (filtered)
Reading program jar [E:\project\hello-scala-master\target\android\intermediates\classes.jar] (filtered)
Reading library jar [C:\Users\DELL\AppData\Local\Android\Sdk\platforms\android-26\android.jar]
Initializing...
java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
        at java.util.concurrent.FutureTask.report(Unknown Source)
        at java.util.concurrent.FutureTask.get(Unknown Source)
        at sbt.ConcurrentRestrictions$$anon$4.take(ConcurrentRestrictions.scala:188)
        at sbt.Execute.next$1(Execute.scala:85)
        at sbt.Execute.processAll(Execute.scala:88)
        at sbt.Execute.runKeep(Execute.scala:68)
        at sbt.EvaluateTask$.liftedTree1$1(EvaluateTask.scala:359)
        at sbt.EvaluateTask$.run$1(EvaluateTask.scala:358)
        at sbt.EvaluateTask$.runTask(EvaluateTask.scala:378)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:69)
        at sbt.Aggregation$$anonfun$3.apply(Aggregation.scala:67)
        at sbt.EvaluateTask$.withStreams(EvaluateTask.scala:314)
        at sbt.Aggregation$.timedRun(Aggregation.scala:67)
        at sbt.Aggregation$.runTasks(Aggregation.scala:76)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:117)
        at sbt.Aggregation$$anonfun$applyDynamicTasks$1.apply(Aggregation.scala:115)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Command$$anonfun$applyEffect$2$$anonfun$apply$3.apply(Command.scala:61)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:253)
        at sbt.Act$$anonfun$sbt$Act$$actParser0$1$$anonfun$sbt$Act$$anonfun$$evaluate$1$1$$anonfun$apply$10.apply(Act.scala:250)
        at sbt.Command$.process(Command.scala:93)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1$$anonfun$apply$1.apply(MainLoop.scala:96)
        at sbt.State$$anon$1.doX$1(State.scala:183)
        at sbt.State$$anon$1.process(State.scala:190)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.MainLoop$$anonfun$1.apply(MainLoop.scala:96)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.MainLoop$.next(MainLoop.scala:96)
        at sbt.MainLoop$.run(MainLoop.scala:89)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:68)
        at sbt.MainLoop$$anonfun$runWithNewLog$1.apply(MainLoop.scala:63)
        at sbt.Using.apply(Using.scala:24)
        at sbt.MainLoop$.runWithNewLog(MainLoop.scala:63)
        at sbt.MainLoop$.runAndClearLast(MainLoop.scala:46)
        at sbt.MainLoop$.runLoggedLoop(MainLoop.scala:30)
        at sbt.MainLoop$.runLogged(MainLoop.scala:22)
        at sbt.StandardMain$.runManaged(Main.scala:109)
        at sbt.xMain.run(Main.scala:38)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
        at xsbt.boot.Launch$.run(Launch.scala:109)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
        at xsbt.boot.Launch$.launch(Launch.scala:117)
        at xsbt.boot.Launch$.apply(Launch.scala:18)
        at xsbt.boot.Boot$.runImpl(Boot.scala:41)
        at xsbt.boot.Boot$.main(Boot.scala:17)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.OutOfMemoryError: GC overhead limit exceeded
[error] java.util.concurrent.ExecutionException: java.lang.OutOfMemoryError: GC overhead limit exceeded
[error] Use 'last' for the full log.
>

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,0 +1,41 @@
+version: 2
+jobs:
+  build:
+    working_directory: /app/
+    docker:
+      - image: docker:17.11.0-ce-rc2-git
+    steps:
+      - checkout
+      - setup_remote_docker
+      - restore_cache:
+          keys:
+            - v1-{{ .Branch }}
+          paths:
+            - /caches/hello-scala.tar
+      - run:
+          name: Load Docker image layer cache
+          command: |
+            set +o pipefail
+            docker load -i /caches/hello-scala.tar | true
+      - run:
+          name: Build docker images
+          command: |
+            docker build --cache-from=hello-scala -t hello-scala .
+      - run:
+          name: Save Docker image layer cache
+          command: |
+            mkdir -p /caches/
+            docker save -o /caches/hello-scala.tar hello-scala
+      - save_cache:
+          key: v1-{{ .Branch }}-{{ epoch }}
+          paths:
+            - /caches/hello-scala.tar
+      - run:
+          name: Run tests
+          command: |
+            docker create -v /app/ --name app alpine /bin/true
+            docker cp /app/ app:/
+            docker run \
+              --volumes-from app \
+              --entrypoint="./entrypoint.sh" \
+              hello-scala

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,5 @@
+.idea/
+.git/
+target/
+project/project/
+project/target/

--- a/.scalafmt
+++ b/.scalafmt
@@ -1,0 +1,1 @@
+maxColumn = 80

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,0 +1,51 @@
+FROM        openjdk:8-jdk-alpine
+
+ENV         ANDROID_BUILD_TOOLS 27.0.0
+ENV         ANDROID_PLATFORM 27
+ENV         ANDROID_SDK 3859397
+ENV         GLIBC "2.25-r0"
+
+# Install wget and unzip; add SSL certs to wget
+RUN         apk update
+RUN         apk add --no-cache ca-certificates wget bash
+RUN         update-ca-certificates
+
+# Install glibc for Android SDK
+RUN         wget -q https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub -O /etc/apk/keys/sgerrand.rsa.pub
+RUN         wget -q https://github.com/sgerrand/alpine-pkg-glibc/releases/download/$GLIBC/glibc-$GLIBC.apk -O /tmp/glibc.apk
+RUN         wget -q https://github.com/sgerrand/alpine-pkg-glibc/releases/download/$GLIBC/glibc-bin-$GLIBC.apk -O /tmp/glibc-bin.apk
+RUN         apk add --no-cache /tmp/glibc.apk /tmp/glibc-bin.apk
+RUN         rm /tmp/glibc.apk /tmp/glibc-bin.apk /etc/apk/keys/sgerrand.rsa.pub
+
+# Install Sbt
+RUN         wget -q https://raw.githubusercontent.com/paulp/sbt-extras/master/sbt -O /bin/sbt
+RUN         chmod +x /bin/sbt
+
+# Install Android SDK
+RUN         mkdir /android-sdk
+RUN         wget -q -O /android-sdk/android-tools.zip https://dl.google.com/android/repository/sdk-tools-linux-$ANDROID_SDK.zip
+RUN         cd /android-sdk && unzip -q ./android-tools.zip
+RUN         rm /android-sdk/android-tools.zip
+ENV         ANDROID_HOME /android-sdk/
+ENV         PATH $PATH:$ANDROID_HOME/tools/:$ANDROID_HOME/platform-tools/
+
+# Install major Android SDK components (tools/bin/sdkmanager --list)
+RUN         yes | /android-sdk/tools/bin/sdkmanager \
+                "platform-tools" \
+                "tools" \
+                "build-tools;$ANDROID_BUILD_TOOLS" \
+                "platforms;android-$ANDROID_PLATFORM" \
+                "extras;android;m2repository" \
+                "extras;google;m2repository"
+
+RUN         mkdir -p /cache/project/
+RUN         mkdir -p /cache/src/main/
+RUN         mkdir -p /cache/src/test/scala/
+ADD         ./build.sbt /cache/
+ADD         ./project/build.properties ./project/plugins.sbt /cache/project/
+RUN         echo "class Test" > /cache/src/test/scala/Test.scala
+RUN         echo "<manifest package=\"com.example\" />" > /cache/src/main/AndroidManifest.xml
+RUN         cd /cache/ && sbt test
+RUN         rm -rf ./cache/
+
+WORKDIR     /app/

--- a/build.sbt
+++ b/build.sbt
@@ -9,7 +9,7 @@ javacOptions ++=
     Nil
 
 libraryDependencies ++=
-    "io.taig" %% "communicator" % "3.3.0-RC1" ::
+    "io.taig" %% "communicator" % "3.3.0-RC2" ::
     "com.android.support" % "appcompat-v7" % "25.3.1" ::
     "com.android.support" % "cardview-v7" % "25.3.1" ::
     "com.android.support" % "design" % "25.3.1" ::

--- a/build.sbt
+++ b/build.sbt
@@ -76,7 +76,7 @@ scalacOptions ++=
     Nil
 
 // Don't upgrade to 2.12.x as it requires Java 8 which does not work with Android
-scalaVersion := "2.11.11"
+scalaVersion := "2.11.12"
 
 targetSdkVersion := "27"
 

--- a/build.sbt
+++ b/build.sbt
@@ -83,3 +83,5 @@ targetSdkVersion := "27"
 versionCode := Some( 0 )
 
 versionName := Some( "0.0.0" )
+
+parallelExecution in Global := false

--- a/build.sbt
+++ b/build.sbt
@@ -41,6 +41,11 @@ platformTarget := "android-27"
 
 proguardCache ++=
     "android.support" ::
+    "cats" ::
+    "io.circe" ::
+    "monix" ::
+    "okhttp3" ::
+    "shapeless" ::
     Nil
 
 proguardOptions ++=

--- a/build.sbt
+++ b/build.sbt
@@ -10,12 +10,12 @@ javacOptions ++=
 
 libraryDependencies ++=
     "io.taig" %% "communicator" % "3.3.0-RC2" ::
-    "com.android.support" % "appcompat-v7" % "25.3.1" ::
-    "com.android.support" % "cardview-v7" % "25.3.1" ::
-    "com.android.support" % "design" % "25.3.1" ::
-    "com.android.support" % "gridlayout-v7" % "25.3.1" ::
-    "com.android.support" % "recyclerview-v7" % "25.3.1" ::
-    "com.android.support" % "support-v4" % "25.3.1" ::
+    "com.android.support" % "appcompat-v7" % "27.0.0" ::
+    "com.android.support" % "cardview-v7" % "27.0.0" ::
+    "com.android.support" % "design" % "27.0.0" ::
+    "com.android.support" % "gridlayout-v7" % "27.0.0" ::
+    "com.android.support" % "recyclerview-v7" % "27.0.0" ::
+    "com.android.support" % "support-v4" % "27.0.0" ::
     "com.google.code.findbugs" % "jsr305" % "3.0.2" % "compile" ::
     "com.geteit" %% "robotest" % "0.12" % "test" ::
     "org.scalatest" %% "scalatest" % "3.0.4" % "test" ::
@@ -37,7 +37,7 @@ packagingOptions := PackagingOptions(
         Nil
 )
 
-platformTarget := "android-26"
+platformTarget := "android-27"
 
 proguardCache ++=
     "android.support" ::
@@ -58,6 +58,8 @@ proguardOptions ++=
     "-dontwarn org.slf4j.**" ::
     Nil
 
+resolvers += "Google Maven" at "https://maven.google.com"
+
 // Shortcut: allows you to execute "sbt run" instead of "sbt android:run"
 run := ( run in Android ).evaluated
 
@@ -71,7 +73,7 @@ scalacOptions ++=
 // Don't upgrade to 2.12.x as it requires Java 8 which does not work with Android
 scalaVersion := "2.11.11"
 
-targetSdkVersion := "26"
+targetSdkVersion := "27"
 
 versionCode := Some( 0 )
 

--- a/build.sbt
+++ b/build.sbt
@@ -18,7 +18,7 @@ libraryDependencies ++=
     "com.android.support" % "support-v4" % "25.3.1" ::
     "com.google.code.findbugs" % "jsr305" % "3.0.2" % "compile" ::
     "com.geteit" %% "robotest" % "0.12" % "test" ::
-    "org.scalatest" %% "scalatest" % "3.0.3" % "test" ::
+    "org.scalatest" %% "scalatest" % "3.0.4" % "test" ::
     Nil
 
 name := "hello-scala"

--- a/build.sbt
+++ b/build.sbt
@@ -1,15 +1,15 @@
-enablePlugins( AndroidApp )
+enablePlugins(AndroidApp)
 
 fork in Test := true
 
 // Enforce Java 7 compilation (in case you have the JDK 8 installed)
 javacOptions ++=
-    "-source" :: "1.7" ::
+  "-source" :: "1.7" ::
     "-target" :: "1.7" ::
     Nil
 
 libraryDependencies ++=
-    "io.taig" %% "communicator" % "3.3.0-RC2" ::
+  "io.taig" %% "communicator" % "3.3.0" ::
     "com.android.support" % "appcompat-v7" % "27.0.0" ::
     "com.android.support" % "cardview-v7" % "27.0.0" ::
     "com.android.support" % "design" % "27.0.0" ::
@@ -29,18 +29,18 @@ minSdkVersion := "14"
 // Prevent common com.android.builder.packaging.DuplicateFileException.
 // Add further file names if you experience the exception after adding new dependencies
 packagingOptions := PackagingOptions(
-    excludes =
-        "META-INF/LICENSE" ::
-        "META-INF/LICENSE.txt" ::
-        "META-INF/NOTICE" ::
-        "META-INF/NOTICE.txt" ::
-        Nil
+  excludes =
+    "META-INF/LICENSE" ::
+      "META-INF/LICENSE.txt" ::
+      "META-INF/NOTICE" ::
+      "META-INF/NOTICE.txt" ::
+      Nil
 )
 
 platformTarget := "android-27"
 
 proguardCache ++=
-    "android.support" ::
+  "android.support" ::
     "cats" ::
     "io.circe" ::
     "monix" ::
@@ -49,7 +49,7 @@ proguardCache ++=
     Nil
 
 proguardOptions ++=
-    "-keepattributes EnclosingMethod,InnerClasses,Signature" ::
+  "-keepattributes EnclosingMethod,InnerClasses,Signature" ::
     "-dontwarn org.w3c.dom.bootstrap.DOMImplementationRegistry" ::
     "-dontwarn javax.xml.bind.DatatypeConverter" ::
     "-dontnote org.joda.time.DateTimeZone" ::
@@ -66,11 +66,11 @@ proguardOptions ++=
 resolvers += "Google Maven" at "https://maven.google.com"
 
 // Shortcut: allows you to execute "sbt run" instead of "sbt android:run"
-run := ( run in Android ).evaluated
+run := (run in Android).evaluated
 
 scalacOptions ++=
-    // Print detailed deprecation warnings to the console
-    "-deprecation" ::
+  // Print detailed deprecation warnings to the console
+  "-deprecation" ::
     // Print detailed feature warnings to the console
     "-feature" ::
     Nil
@@ -80,8 +80,8 @@ scalaVersion := "2.11.11"
 
 targetSdkVersion := "27"
 
-versionCode := Some( 0 )
+versionCode := Some(0)
 
-versionName := Some( "0.0.0" )
+versionName := Some("0.0.0")
 
 parallelExecution in Global := false

--- a/circle.yml
+++ b/circle.yml
@@ -6,8 +6,7 @@ dependencies:
   pre:
     - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.16.deb
     - sudo dpkg -i sbt-0.13.16.deb
-    - ( sleep 5 && while [ 1 ]; do sleep 1; echo y; done ) | android update sdk --no-ui --all --filter "tools,platform-tools,android-27,extra-google-m2repository,extra-android-m2repository,extra-google-google_play_services"
-    - echo y | android update sdk --no-ui --all --filter "build-tools-27.0.0"
+    - echo y | android update sdk --no-ui --all --filter "tools,android-27,build-tools-27.0.0,platform-tools"
   cache_directories:
     - "~/.ivy2"
     - "~/.sbt"

--- a/circle.yml
+++ b/circle.yml
@@ -4,10 +4,10 @@ machine:
 
 dependencies:
   pre:
-    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.15.deb
-    - sudo dpkg -i sbt-0.13.15.deb
-    - ( sleep 5 && while [ 1 ]; do sleep 1; echo y; done ) | android update sdk --no-ui --all --filter "tools,platform-tools,android-26,extra-google-m2repository,extra-android-m2repository,extra-google-google_play_services"
-    - echo y | android update sdk --no-ui --all --filter "build-tools-25.0.3"
+    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.16.deb
+    - sudo dpkg -i sbt-0.13.16.deb
+    - ( sleep 5 && while [ 1 ]; do sleep 1; echo y; done ) | android update sdk --no-ui --all --filter "tools,platform-tools,android-27,extra-google-m2repository,extra-android-m2repository,extra-google-google_play_services"
+    - echo y | android update sdk --no-ui --all --filter "build-tools-27.0.0"
   cache_directories:
     - "~/.ivy2"
     - "~/.sbt"

--- a/circle.yml
+++ b/circle.yml
@@ -1,18 +1,11 @@
 machine:
-  java:
-    version: oraclejdk8
+  services: docker
 
 dependencies:
-  pre:
-    - wget -q https://dl.bintray.com/sbt/debian/sbt-0.13.16.deb
-    - sudo dpkg -i sbt-0.13.16.deb
-    - echo y | android update sdk --no-ui --all --filter "tools,android-27,build-tools-27.0.0,platform-tools"
-  cache_directories:
-    - "~/.ivy2"
-    - "~/.sbt"
   override:
-    - sbt test:compile
+    - docker info
+    - docker build -t hello-scala .  
 
 test:
   override:
-    - sbt test
+    - docker run hello-scala

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,4 +1,6 @@
 #!/bin/bash
 
-sbt "scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
+set -e
+
+sbt ";scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
 sbt test

--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -1,0 +1,4 @@
+#!/bin/bash
+
+sbt "scalafmt::test;test:scalafmt::test;sbt:scalafmt::test"
+sbt test

--- a/project/build.properties
+++ b/project/build.properties
@@ -1,1 +1,1 @@
-sbt.version=0.13.15
+sbt.version=0.13.16

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -1,3 +1,3 @@
 addSbtPlugin( "io.taig" % "sbt-scalariform" % "1.7.1" )
 
-addSbtPlugin( "org.scala-android" % "sbt-android" % "1.7.7" )
+addSbtPlugin( "org.scala-android" % "sbt-android" % "1.7.10" )

--- a/project/plugins.sbt
+++ b/project/plugins.sbt
@@ -1,3 +1,3 @@
-addSbtPlugin( "io.taig" % "sbt-scalariform" % "1.7.1" )
+addSbtPlugin( "com.lucidchart" % "sbt-scalafmt" % "1.14" )
 
 addSbtPlugin( "org.scala-android" % "sbt-android" % "1.7.10" )

--- a/src/main/scala/com/example/Hello.scala
+++ b/src/main/scala/com/example/Hello.scala
@@ -5,23 +5,23 @@ import android.support.v7.app.AppCompatActivity
 import android.view.MenuItem
 
 class Hello extends AppCompatActivity {
-    override def onCreate( savedInstanceState: Bundle ): Unit = {
-        super.onCreate( savedInstanceState )
+  override def onCreate(savedInstanceState: Bundle): Unit = {
+    super.onCreate(savedInstanceState)
 
-        val view: TypedViewHolder.main =
-            TypedViewHolder.setContentView( this, TR.layout.main )
+    val view: TypedViewHolder.main =
+      TypedViewHolder.setContentView(this, TR.layout.main)
 
-        setSupportActionBar( view.toolbar )
-        getSupportActionBar.setDisplayHomeAsUpEnabled( true )
+    setSupportActionBar(view.toolbar)
+    getSupportActionBar.setDisplayHomeAsUpEnabled(true)
 
-        view.welcome.setText( R.string.name )
+    view.welcome.setText(R.string.name)
+  }
+
+  override def onOptionsItemSelected(item: MenuItem): Boolean =
+    item.getItemId match {
+      case android.R.id.home ⇒
+        finish()
+        true
+      case _ ⇒ super.onOptionsItemSelected(item)
     }
-
-    override def onOptionsItemSelected( item: MenuItem ): Boolean =
-        item.getItemId match {
-            case android.R.id.home ⇒
-                finish()
-                true
-            case _ ⇒ super.onOptionsItemSelected( item )
-        }
 }

--- a/src/main/scala/com/example/Hello.scala
+++ b/src/main/scala/com/example/Hello.scala
@@ -4,22 +4,24 @@ import android.os.Bundle
 import android.support.v7.app.AppCompatActivity
 import android.view.MenuItem
 
-class Hello extends AppCompatActivity with TypedFindView {
+class Hello extends AppCompatActivity {
     override def onCreate( savedInstanceState: Bundle ): Unit = {
         super.onCreate( savedInstanceState )
 
-        setContentView( R.layout.main )
+        val view: TypedViewHolder.main =
+            TypedViewHolder.setContentView( this, TR.layout.main )
 
-        setSupportActionBar( findView( TR.toolbar ) )
+        setSupportActionBar( view.toolbar )
         getSupportActionBar.setDisplayHomeAsUpEnabled( true )
 
-        findView( TR.welcome ).setText( R.string.name )
+        view.welcome.setText( R.string.name )
     }
 
-    override def onOptionsItemSelected( item: MenuItem ) = item.getItemId match {
-        case android.R.id.home ⇒
-            finish()
-            true
-        case _ ⇒ super.onOptionsItemSelected( item )
-    }
+    override def onOptionsItemSelected( item: MenuItem ): Boolean =
+        item.getItemId match {
+            case android.R.id.home ⇒
+                finish()
+                true
+            case _ ⇒ super.onOptionsItemSelected( item )
+        }
 }

--- a/src/test/scala/com/example/test/Test.scala
+++ b/src/test/scala/com/example/test/Test.scala
@@ -4,14 +4,11 @@ import android.os.Build.VERSION_CODES.LOLLIPOP
 import com.example.R
 import org.robolectric.RuntimeEnvironment
 import org.robolectric.annotation.Config
-import org.scalatest.{ FlatSpec, Matchers, RobolectricSuite }
+import org.scalatest.{FlatSpec, Matchers, RobolectricSuite}
 
-@Config( sdk = Array( LOLLIPOP ) )
-class Test
-        extends FlatSpec
-        with Matchers
-        with RobolectricSuite {
-    "Resources" should "be accessible via R" in {
-        RuntimeEnvironment.application.getString( R.string.name ) shouldBe "Hello Scala!"
-    }
+@Config(sdk = Array(LOLLIPOP))
+class Test extends FlatSpec with Matchers with RobolectricSuite {
+  "Resources" should "be accessible via R" in {
+    RuntimeEnvironment.application.getString(R.string.name) shouldBe "Hello Scala!"
+  }
 }

